src/Package/Impl/Options/R/RToolsOptionsPage.cs

```diff
@@ -198,19 +198,15 @@ public class RToolsOptionsPage : DialogPage {
         [LocCategory("Settings_LogCategory")]
         [CustomLocDisplayName("Settings_LogLevel")]
         [LocDescription("Settings_LogLevel_Description")]
-        [TypeConverter(typeof(LogLevelTypeConverter))]
-#if DEBUG
-        [DefaultValue(LogVerbosity.Traffic)]
-#else
+        [TypeConverter(typeof(LogVerbosityTypeConverter))]
         [DefaultValue(LogVerbosity.Normal)]
-#endif
         public LogVerbosity LogVerbosity {
             get {
-#if DEBUG
-                return _holder.GetValue<LogVerbosity>(LogVerbosity.Traffic);
-#else
-                return _holder.GetValue<LogVerbosity>(LogVerbosity.Normal);
-#endif
+                var permissions = VsAppShell.Current.Services.LoggingPermissions;
+                var value = _holder.GetValue<LogVerbosity>(permissions.MaxVerbosity);
+                value = MathExtensions.Min<LogVerbosity>(value, permissions.MaxVerbosity);
+                _holder.SetValue(value);
```

Comment:
Doesn't really understand this piece. Shouldn't we restrict max value in setter?